packages/clerk-js/src/ui/elements/Menu.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { PropsWithChildren, useEffect, useRef } from 'react';
+
+import { createContextAndHook } from '../../ui/utils';
+import { Button, Col } from '../customizables';
+import { usePopover } from '../hooks';
+import { animations, PropsOfComponent } from '../styledSystem';
+import { colors } from '../utils/colors';
+import { Portal } from './Portal';
+
+type UsePopoverReturn = ReturnType<typeof usePopover>;
+
+type MenuState = {
+  popoverCtx: UsePopoverReturn;
+};
+
+const [MenuStateCtx, useMenuState] = createContextAndHook<MenuState>('MenuState');
+
+type MenuProps = PropsWithChildren<Record<never, never>>;
+
+export const Menu = (props: MenuProps) => {
+  const popoverCtx = usePopover({
+    placement: 'right-start',
+    offset: 8,
+  });
+
+  return (
+    <MenuStateCtx.Provider
+      value={{ value: { popoverCtx } }}
```

Comment:
`{ value: { popoverCtx } }` will get create a new object reference on each render, causing all other listeners of that context to rerender as a result.

Lets use a `useMemo` as we do in other places as well.

Reply:
✔️